ccl/sqlproxyccl: TestDenylistUpdate failed

ccl/sqlproxyccl.TestDenylistUpdate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11430905?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11430905?buildTab=artifacts#/) on release-22.2 @ [42fa1fad9af04139575e4f374a8b3f2d4ca453b1](https://github.com/cockroachdb/cockroach/commits/42fa1fad9af04139575e4f374a8b3f2d4ca453b1):


```
=== RUN   TestDenylistUpdate
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/a455c59caed0cf9d0a0298ad88bfe7a2/logTestDenylistUpdate238667709
    test_log_scope.go:79: use -show-logs to present logs inline
    proxy_handler_test.go:827: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/proxy_handler_test.go:827
        	Error:      	Expect "read tcp 127.0.0.1:43828->127.0.0.1:46727: read: connection reset by peer" to match "closed|bad connection"
        	Test:       	TestDenylistUpdate
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a455c59caed0cf9d0a0298ad88bfe7a2/logTestDenylistUpdate238667709
--- FAIL: TestDenylistUpdate (1.99s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDenylistUpdate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30830

I managed to reproduce this with 5K runs:

```
==================== Test output for //pkg/ccl/sqlproxyccl:sqlproxyccl_test (run 1644 of 5000):
initialized metamorphic constant "span-reuse-rate" with value 76
initialized metamorphic constant "parse-json-impl" with value 0
initialized metamorphic constant "coldata-batch-size" with value 1565
initialized metamorphic constant "mvcc-incremental-iter-tbi" with value false
initialized metamorphic constant "mvcc-value-disable-simple-encoding" with value true
initialized metamorphic constant "mvcc-max-iters-before-seek" with value 0
initialized metamorphic constant "disable-checksstconflicts-range-key-masking" with value true
initialized metamorphic constant "kv.rangefeed.use_dedicated_connection_class.enabled" with value true
initialized metamorphic constant "kv-batch-size" with value 1
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "inverted-joiner-batch-size" with value 1
initialized metamorphic constant "zig-zag-joiner-batch-size" with value 1
initialized metamorphic constant "merge-joiner-groups-buffer" with value 8
initialized metamorphic constant "direct-scans-enabled" with value true
initialized metamorphic constant "ColIndexJoin-batch-size" with value 622204
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 7540247
initialized metamorphic constant "max-batch-size" with value 8646
initialized metamorphic constant "max-batch-byte-size" with value 27284002
initialized metamorphic constant "parallel-scan-result-threshold" with value 6470
initialized metamorphic constant "resolve-membership-single-scan-enabled" with value false
initialized metamorphic constant "copy-batch-size" with value 6745
initialized metamorphic constant "write-metadata-sst" with value true
initialized metamorphic constant "backup.write_metadata_with_external_ssts.enabled" with value false
initialized metamorphic constant "raft-log-truncation-clearrange-threshold" with value 450904
initialized metamorphic constant "raft-step-down-on-removal" with value false
initialized metamorphic constant "backup-read-manifest" with value true
initialized metamorphic constant "export_request_verbose_tracing" with value true
initialized metamorphic constant "changefeed.event_consumer_worker_queue_size" with value 1
initialized metamorphic constant "changefeed.new_webhook_sink.enabled" with value false
initialized metamorphic constant "aws-reuse-kms" with value false
I230829 02:49:52.901521 1 (gostd) rand.go:249  [-] 1  random seed: -2402519637672264509
--- FAIL: TestDenylistUpdate (4.58s)
...
    proxy_handler_test.go:1259:
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/proxy_handler_test.go:1259
        	Error:      	Expect "read tcp 127.0.0.1:56277->127.0.0.1:56275: read: connection reset by peer" to match "closed|bad connection"
        	Test:       	TestDenylistUpdate
    panic.go:522: -- test log scope end --
```

--- a/pkg/ccl/sqlproxyccl/proxy_handler_test.go
+++ b/pkg/ccl/sqlproxyccl/proxy_handler_test.go
@@ -1256,7 +1256,7 @@ func TestDenylistUpdate(t *testing.T) {
 		"Expected the connection to eventually fail",
 	)
 	require.Error(t, err)
-	require.Regexp(t, "closed|bad connection", err.Error())
+	require.Regexp(t, "(connection reset by peer|closed|bad connection)", err.Error())
 	require.Equal(t, int64(1), s.metrics.ExpiredClientConnCount.Count())
 }
 
@@ -2279,7 +2279,7 @@ func TestConnectionMigration(t *testing.T) {
 			t.Fatalf("require that pg_sleep query terminates")
 		case err = <-errCh:
 			require.Error(t, err)
-			require.Regexp(t, "(closed|bad connection)", err.Error())
+			require.Regexp(t, "(connection reset by peer|closed|bad connection)", err.Error())
 		}
 
 		require.EqualError(t, f.ctx.Err(), context.Canceled.Error())

